notebooks/nb_utils/lesson_preprocessor.py

```diff
@@ -61,14 +61,15 @@ def preprocess(self, nb, resources):
         return nb, resources
 
     def add_header_and_footer(self, cells):
-        """Returns a list of cells with a header cell first, followed by cells (which is a list), followed by a footer cell"""
-        course_info ="""**[{} Course Home Page]({})**""".format(self.track.course_name, self.track.course_url)
- 
-        header_content = course info + "\n---"
-        footer_content = "---\n" + course_info
-        header_cell = make_cell(cell_type='markdown', source=header_content)
-        footer_cell = make_cell(cell_type='markdown', source=footer_content)
-        return header_cell + cells + footer_cells
+        """Inserts header cell at front of cells and appends footer cell to end. Both new cells have course links"""
+        course_link ="""**[{} Course Home Page]({})**\n\n""".format(self.track.course_name, self.track.course_url)
+        horizontal_line_break = "---\n"
```

Comment:
This does change the formatting slightly from the current version; however, the layout is different anyway due to being in it's own cell.  As long as you're happy with the way it looks, it should be fine.